### PR TITLE
Add Hadolint Dockerfile Scan

### PR DESCRIPTION
**Pull request by jandradap**
# Changes

This PR introduces tasks for Hadolint.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/hadolint/0.1/hadolint.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        The output format for the results [tty | json | checkstyle | codeclimate
+        | gitlab_codeclimate | codacy] (default tty).
+      name: output-format
+      type: string
+  steps:
+    - image: 'ghcr.io/hadolint/hadolint:v2.8.0-alpine'
+      name: lint-dockerfile
+      resources: {}
+      script: |
+        ls -lh
```
Comment (chmouel) [inline task/hadolint/0.1/hadolint.yaml:40]: is that a debug statement?

Comment (chmouel) [reply task/hadolint/0.1/hadolint.yaml:40]: maybe you want to add a "set -e" so if the linter fail the task will fail too,

Comment (jandradap) [reply task/hadolint/0.1/hadolint.yaml]: added "set -e" and delete the debug statement "ls -lh"

File: task/hadolint/0.1/tests/run.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  params:
+  - name: repository
+    value: https://github.com/docker-library/hello-world.git
+  - name: ignore-rules
+    value: 'DL3007'
+  - name: dockerfile-path
+    value: './amd64/hello-world/Dockerfile'
+  workspaces:
+    - name: source
+      persistentvolumeclaim:
```
Comment (vinamra28) [inline task/hadolint/0.1/tests/run.yaml:59]: nit:
```suggestion
      persistentVolumeClaim:
```

File: task/hadolint/0.1/samples/run.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  params:
+  - name: repository
+    value: https://github.com/docker-library/hello-world.git
+  - name: ignore-rules
+    value: 'DL3007'
+  - name: dockerfile-path
+    value: './amd64/hello-world/Dockerfile'
+  workspaces:
+    - name: source
+      persistentvolumeclaim:
```
Comment (vinamra28) [inline task/hadolint/0.1/samples/run.yaml:59]: same as above

File: task/hadolint/0.1/hadolint.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      name: dockerfile-path
+      type: string
+    - default: tty
+      description: >-
+        The output format for the results [tty | json | checkstyle | codeclimate
+        | gitlab_codeclimate | codacy] (default tty).
+      name: output-format
+      type: string
+  steps:
+    - image: 'ghcr.io/hadolint/hadolint:v2.8.0-alpine'
```
Comment (vinamra28) [inline task/hadolint/0.1/hadolint.yaml:36]: can we add a digest to the image as well. example `docker.io/alpine/git:v2.26.2@sha256:23618034b0be9205d9cc0846eb711b12ba4c9b468efdd8a59aac1d7b1a23363f`

File: task/hadolint/0.1/hadolint.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      name: lint-dockerfile
+      resources: {}
```
Comment (vinamra28) [inline task/hadolint/0.1/hadolint.yaml:38]: I guess we can remove this

File: task/hadolint/0.1/samples/run.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: hadolint-pipeline
+spec:
+  params:
+    - name: repository
```
Comment (vinamra28) [inline task/hadolint/0.1/samples/run.yaml:8]: I guess `run.yaml` here and the one in `tests/run.yaml` have same contents. Maybe we can remove `samples` directory as they both are identical

File: task/hadolint/0.1/README.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+kind: Pipeline
+metadata:
+  name: hadolint-pipeline
+spec:
+  params:
+    - name: repository
+      type: string
+    - name: ignore-rules
+      type: string
+    - name: dockerfile-path
```
Comment (vinamra28) [inline task/hadolint/0.1/README.md:52]: similarly here. Probably you can remove this YAML from here and just add a reference to `run.yaml` from tests directory. Maybe you can take some reference from [`aws-cli`](https://github.com/tektoncd/catalog/tree/main/task/aws-cli/0.2#usage) task.

File: task/hadolint/0.1/samples/run.yaml (hunk 3 above)
Comment (jandradap) [reply task/hadolint/0.1/samples/run.yaml:59]: Fixed

File: task/hadolint/0.1/samples/run.yaml (hunk 6 above)
Comment (jandradap) [reply task/hadolint/0.1/samples/run.yaml:8]: Fixed

File: task/hadolint/0.1/hadolint.yaml (hunk 4 above)
Comment (jandradap) [reply task/hadolint/0.1/hadolint.yaml:36]: Fixed.
` ~  podman inspect --format='{{index .RepoDigests 0}}' ghcr.io/hadolint/hadolint:v2.8.0-alpine                                                                                                                                                              ghcr.io/hadolint/hadolint@sha256:6c4b7c23f96339489dd35f21a711996d7ce63047467a9a562287748a03ad5242`

File: task/hadolint/0.1/README.md (hunk 7 above)
Comment (jandradap) [reply task/hadolint/0.1/README.md:52]: Fixed.

File: task/hadolint/0.1/hadolint.yaml (hunk 5 above)
Comment (jandradap) [reply task/hadolint/0.1/hadolint.yaml:38]: removed line 38.

**Review by vinamra28 — COMMENTED** [review]
PR looks good to me. Can you please squash the commits as well?

**Comment by jandradap** [comment]
@vinamra28 I already squashed the commits. Thanks for your review, the next contribution will be more efficient.

**Comment by vinamra28** [comment]
hey, sorry for bothering you again. Can you please follow the commit message best practices present [here](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages) ?

**Comment by jandradap** [comment]
@vinamra28 just modify the commit message and fix the warn:
`WARN : Step "lint-dockerfile" uses image "ghcr.io/hadolint/hadolint@sha256:6c4b7c23f96339489dd35f21a711996d7ce63047467a9a562287748a03ad5242"; consider using a image tagged with specific version along with digest eg. abc.io/img:v1@sha256:abcde`

**Pull request merged**